### DISCUSSION TITLE
How Can I Join 2 Commands?

### DISCUSSION THREAD

**Discussion by 918483** [savonet/liquidsoap#3570] [General]
I would like to turn these 2 commands into 1 only the first would be the song that is playing and the second would be the next song there is a way to join the functions to then trigger using the 
```
harbor.http.register.simple(port=8080, method="GET", path, handler)
```

```
def idnew(m) =

  title = m["title"]
  artist = m["artist"]
  
 [("title","#{title}"),("artist","#{artist}")]
 print("Music Playing #{title}")
  
end

def check(r)

  n = request.metadata(r)
  t = string.split(separator="/",n["filename"])
  next = string.replace(pattern=".mp3", (fun(_) -> ""), list.nth(t,5))
  print("Next Music: #{next}")
  true
  
 end
```

**Comment by vitoyucepi** [comment]
Hi @918483,
Do you want to get the current song and the next song in a single http request?

**Reply by 918483** [reply]
That's right take the 2 info and fire in HTTP request

**Reply by vitoyucepi** [reply]
OK, I'll try to start from the general form to a concrete implementation.
1. You want to serve an http endpoint on `127.0.0.1:8080/api/source_info`.
1. To get the data from liquidsoap you'll use a *GET* request like this `curl http://127.0.0.1:8080/api/source_info`.
1. Since it is a *GET* request, no information will be changed.

Script
```ruby
api_port = 8080
api_path = "/api/source_info"

def source_info_handler(_)
  data = {
    current_song = {title = "a", author = "A"},
    next_song = {title = "b", author = "B"}
  }
  http.response(
    status_code=200,
    content_type="application/json; charset=UTF-8",
    data=json.stringify(data, compact=true) ^ "\n"
  )
end

harbor.http.register.simple(api_path, source_info_handler, port=api_port, method="GET")
```

Request command
```sh
curl http://127.0.0.1:8080/api/source_info
```

Response
```json
{"current_song":{"title":"a","author":"A"},"next_song":{"title":"b","author":"B"}}
```

Does my idea fit into the plan you have?

**Reply by 918483** [reply]
that's right but I don't want to get the data I want to send it via POST to a url that is in another location

**Reply by 918483** [reply]
Take the 2 pieces of information and pass with the function down

curl -s --request POST --url http://web/api/internal/1/djoff --form current_song="info" --form next_song="info"

**Reply by 918483** [reply]
And in this case, here would be showing the song that's playing and the next one

current_song = {title = "a", author = "A"},
next_song = {title = "b", author = "B"}

**Reply by vitoyucepi** [reply]
Are you attempting to modify the behavior of AzuraCast? Is my understanding correct?
